chat/src/user-info/Messages.jsx

```diff
@@ -0,0 +1,293 @@
+  // to the specified Firebase reference. By using these auto-generated keys
+  // for each new element in the list, several clients can add children to the
+  // same location at the same time without write conflicts. The unique key
+  // generated by push() is based on a timestamp, so list items are automatically
+  // ordered chronologically.
+
+  const handleSubmit = async (e) => {
+    var today = new Date();
+    var Time = today.getHours() + ":" + today.getMinutes();
+    e.preventDefault();
+    const db = getDatabase();
+    const firstuserkey = push(
+      ref(db, "Users/Chat/" + user.uid + userDetails.uid)
+    ).key;
+    const sender = ref(
+      db,
+      "Users/Chat/" + user.uid + userDetails.uid + "/" + firstuserkey
+    );
+
+    // sender
+
+    set(sender, {
+      message: content,
+      sender: user.uid + "/" + user.username,
+      receiver: userDetails.uid + "/" + userDetails.username,
+      messageId: firstuserkey,
+      time: Time,
+    })
+      .then((e) => {
+        const receiver = ref(
+          db,
+          "Users/Chat/" + userDetails.uid + user.uid + "/" + firstuserkey
+        );
+
+        // receiver
+
+        set(receiver, {
+          message: content,
+          sender: user.uid + "/" + user.username,
+          receiver: userDetails.uid + "/" + userDetails.username,
+          messageId: firstuserkey,
+          time: Time,
+        });
+      })
+      .then(() => data(user));
+    setContent("");
+  };
+
+  // delete function
+
+  // for sender
+  const deleteForMe = (messageId) => {
+    setShow(false);
+    const keyref = ref(
+      db,
+      "Users/Chat/" + user.uid + userDetails.uid + "/" + messageId
+    );
+    const keyrefs = ref(
+      db,
+      "Users/Chat/" + userDetails.uid + user.uid + "/" + messageId
+    );
+    if (keyref) {
+      remove(keyref);
+    } else if (keyrefs) {
+      remove(keyrefs);
+    }
+  };
+  // for sender and receiver
+  const deleteForEveryone = (messageId) => {
+    setShow(false);
+    const keyref = ref(
+      db,
+      "Users/Chat/" + user.uid + userDetails.uid + "/" + messageId
+    );
+    const keyrefs = ref(
+      db,
+      "Users/Chat/" + userDetails.uid + user.uid + "/" + messageId
+    );
+    remove(keyref).then(() => {
+      remove(keyrefs).then(() => data(user));
+    });
+  };
+
+  // setting key for modal
+
+  const [modalValues, setModalValues] = useState();
+  const modalFun = (senderMessages) => {
+    handleShow();
+    setModalValues(senderMessages.messageId);
+  };
+
+  // show / close modal
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  // scroll messages/page
+
+  const AlwaysScrollToBottom = () => {
+    const elementRef = useRef();
+    useEffect(() => elementRef.current.scrollIntoView());
+    return <div ref={elementRef} />;
+  };
+
+  // navigate back to profile page
+
+  const backToProfile = () => {
+    window.location.href = "/profile";
+  };
+
+  if (loading) {
```

Comment:
Make a generic component of loading which can be imported and used at all the places.